Src/AutoFixtureUnitTest/EmailAddressLocalPartGeneratorTest.cs

```diff
@@ -0,0 +1,165 @@
+    {
+        [Fact]
+        public void SutIsSpecimenBuilder()
+        {
+            // Fixture setup
+            // Exercise system
+            var sut = new EmailAddressLocalPartGenerator();
+            // Verify outcome
+            Assert.IsAssignableFrom<ISpecimenBuilder>(sut);
+            // Teardown
+        }
+
+        [Fact]
+        public void CreateWithNullRequestReturnsCorrectResult()
+        {
+            // Fixture setup
+            var sut = new EmailAddressLocalPartGenerator();
+            // Exercise system
+            var dummyContext = new DelegatingSpecimenContext();
+            var result = sut.Create(null, dummyContext);
+            // Verify outcome
+            Assert.Equal(new NoSpecimen(), result);
+            // Teardown
+        }
+
+        [Fact]
+        public void CreateWithNullContextThrows()
+        {
+            // Fixture setup
+            var sut = new EmailAddressLocalPartGenerator();
+            var dummyRequest = new object();
+            // Exercise system and verify outcome
+            Assert.Throws<ArgumentNullException>(() => sut.Create(dummyRequest, null));
+            // Teardown
+        }
+
+        [Fact]
+        public void CreateWithNonEmailAddressLocalPartRequestReturnsCorrectResult()
+        {
+            // Fixture setup
+            var sut = new EmailAddressLocalPartGenerator();
+            var dummyRequest = new object();
+            // Exercise system
+            var dummyContext = new DelegatingSpecimenContext();
+            var result = sut.Create(dummyRequest, dummyContext);
+            // Verify outcome
+            var expectedResult = new NoSpecimen(dummyRequest);
+            Assert.Equal(expectedResult, result);
+            // Teardown
+        }
+
+        [Fact]
+        public void CreateWhenLocalPartReceivedFromContextIsNullReturnsCorrectResult()
+        {
+            // Fixture setup
+            var request = typeof(EmailAddressLocalPart);
+            object expectedValue = null;
+            var context = new DelegatingSpecimenContext
+            {
+                OnResolve = r => typeof(string).Equals(r) ? expectedValue : new NoSpecimen(r)
+            };
+            var sut = new EmailAddressLocalPartGenerator();
+            // Exercise system and verify outcome
+            var result = sut.Create(request, context);
+            var expectedResult = new NoSpecimen(request);
+            Assert.Equal(expectedResult, result);
+            // Teardown
+        }
+
+        [Fact]
+        public void CreateReturnsEmailAddressLocalPartUsingLocalPartReceivedFromContext()
+        {
+            // Fixture setup
+            var request = typeof(EmailAddressLocalPart);
+            string expectedLocalPart = Guid.NewGuid().ToString();
+            var context = new DelegatingSpecimenContext
+            {
+                OnResolve = r =>
+                {
+                    if (typeof(string).Equals(r))
```

Comment:
You could add a [Guard Assertion](http://xunitpatterns.com/Guard%20Assertion.html) like this:

```C#
var context = new DelegatingSpecimenContext
{
    OnResolve = r =>
    {
        Assert.Equal(typeof(string), r);
        return expectedLocalPart;
    }
};
```

Very theoretically: this actually turns a [Stub](http://xunitpatterns.com/Test%20Stub.html) into a [Mock](http://xunitpatterns.com/Mock%20Object.html), which is [isn't really appropriate](http://blog.ploeh.dk/2013/10/23/mocks-for-commands-stubs-for-queries), but is, IMO, the lesser of two evils, because in this case, the code is easier to reason about, and the Mock is very localised.